### ISSUE TITLE
Change how the addon names are named

### ISSUE THREAD

**Issue by Hans5958** [ScratchAddons/ScratchAddons#2764]
This is a group of suggestion related to the addon names.

- [1. Use consistent grammar](#1-use-consistent-grammar)
- [2. Shorten the names](#2-shorten-the-names)
- [3. Use title case](#3-use title case)

## 1. Use consistent grammar

The addon names are quite inconsistent in terms of the grammar. There are some that are phrases that starts with a verb (view, disable, etc.) and there are some that are just nouns.

Here's the list of a quantity of addons that I tagged based on if the names are phrases (with verb) or nouns.

![firefox_2021-06-13_19-39-57](https://user-images.githubusercontent.com/11584103/121808170-81797400-cc81-11eb-8206-edd0b9f8d15f.png)
![firefox_2021-06-13_19-42-22](https://user-images.githubusercontent.com/11584103/121808174-83dbce00-cc81-11eb-93fc-e7be54a60326.png)
![firefox_2021-06-13_19-44-24](https://user-images.githubusercontent.com/11584103/121808177-850cfb00-cc81-11eb-9aaa-6d7001b31498.png)

Seems that we have more nouns than phrases, we should change them to nouns. (Yes, I know there might some that can't be made as a noun, but just keep in minimum)

## 2. Shorten the names

Some names are just ridiculously long for just a name. Here are some examples.

- Filter messages on Scratch's messages page 
- Change new sprite default position 
- Customized quotes & code blocks on forums
- Show ocular.jeffalo.net statuses and post reactions

Again, this is quite ridiculous! My suggestion: Make them short, and put additional context inside the description. (Again, I know there might some that can, but at least we should try!)

## 3. Use title case

While we're at it, we should use title case (Feature Name) instead of sentence case (Feature name). This may be a mistake since I suggest to use sentence case, since last time there were more names with phrases than nouns, but looking that we have more names that just outright as a feature name (and a noun), we should just make them title case.

I mean, just look how other people do it. It's way better to use title case!

![firefox_2021-06-13_19-46-10](https://user-images.githubusercontent.com/11584103/121808524-1e88dc80-cc83-11eb-9054-c02e0e18e76a.png)
![firefox_2021-06-13_19-48-32](https://user-images.githubusercontent.com/11584103/121808526-1fba0980-cc83-11eb-92bd-5915b47955af.png)
![firefox_2021-06-13_19-50-19](https://user-images.githubusercontent.com/11584103/121808528-20eb3680-cc83-11eb-854f-44816f614a9a.png)
![firefox_2021-06-13_19-50-46](https://user-images.githubusercontent.com/11584103/121808530-22b4fa00-cc83-11eb-9e13-7c05e54282b7.png)
![Discord_2021-06-13_20-07-10](https://user-images.githubusercontent.com/11584103/121808532-234d9080-cc83-11eb-9e44-4374021cd124.png)

**Comment by Shluffy** [comment]
what is wrong with having nouns and phrases?

**Comment by apple502j** [comment]
Finally, after more than half a year:

![image](https://user-images.githubusercontent.com/33279053/121810100-6bc47880-cc9a-11eb-8f3f-839a169acfce.png)

**Comment by mxmou** [comment]
Title case is harder to read, especially in lists, even if the names are short. Being a noun or not doesn't change anything.

**Comment by Hans5958** [comment]
> what is wrong with having nouns and phrases?

Consistency is my main point, but there might other reasons why. One I think is that it makes the names longer.

> Finally, after more than half a year:
>
> [image]

That's misleading. I complained because you took the matter to yourself (read: coup). I discussed this matter twice, here and a discussion on the Discussion tab, without doing it myself (even though I could just do it, I have push access). 

Even so, the consensus a while ago agreed to do this, and I think it's time to look back and think again about this decision and reach a new consensus.

**Comment by Hans5958** [comment]
> Title case is harder to read, especially in lists.

I don't think that title case will affect readability. Even so, look at my images as examples.

> Being a noun or not doesn't change anything.

Yes, I agree, but my main point here is to be consistent.

**Comment by apple502j** [comment]
As a Japanese translator POV:
- title case doesn't matter because ~~we are communists~~ there is no capitals
- name length sometimes matter but sometimes don't. because of how we use ideographic letters (hm, why is my autocorrect saying "demographic" is the correct word) for nouns it can be shortened in most cases. Exceptions exist though.

Japanese addon names that are noun:
2D カラーピッカー (2d color picker), 60FPSモード (60 fps mode. yes, i omitted "alt-click" part intentionally), 何もしないアドオン (addon that does nothing), サムネイル設定 (thumbnail configuration), プロフィールページバナー (profile page banner), ブロックパレットアイコン (block palette icons), ブロック置換 (block replacements), キャットブロック (cat block), クローンカウンター (clone counter), クラウドゲーム (cloud game), 16進数カラーピッカー (hex color picker), コメント改行 (comment linebreak), 確認画面 (confirmation screen, yes, 4 letters.), キュレーターリンク (curator link), カスタムズーム設定 (custom zoom settings), 団子 (エイプリルフール機能) (dango (april fools feature)), ウェブサイトダークモード (website darkmode), 変数ブロック改良 (variable block improvement), デバッガー (debugger), 話すボタン (discuss button), エディターのダークモードとカスタマイズ (editor darkmode and customization), 開発者ツール (developer tools), ドロップダウン検索 (dropdown search), エディター効果音 (editor sound effects), 伸縮可能検索バー (expandable search bar), スプライトフォルダー (sprite folder), フォーラム検索 (forum search), ゲームパッド (gamepad), フォーラム画像アップローダー (forum image uploader), 無限スクロール (infinite scroll), 共有・編集日表示 (shared/edited date display), メッセージフィルター (message filter), マウスの座標 (mouse positions), 半透明コスチュームエディター (translucent costume editor), 一時停止ボタン (pause button), 読み込み進捗バー (loading progress bar), スプライト・スクリプト数表示 (sprite/script count display), 使い方タブ (usage tab), リミックスツリーボタン (remix tree button), Scratchメッセージ (scratch messages), Scratch通知 (scratch notifications), プロフィール統計 (profile stats), プロフィール検索 (profile search), セミコロンのバグ (semicolon bug), スタジオ管理ツール (studio management tools), 透明ブロック (transparent block ~~like glass, glowstone, leaves~~), 変数マネージャー (variable manager), YouTube全画面表示 (youtube fullscreen display)

and phrases?:
絵文字の変更 (changing emojis), 選択範囲を引用 (quoting selected range), ビットマップ画像をコピー (copying bitmap images), ブロックを画像に出力 (outputting blocks to images), コメントのリンクをコピー (copying comment links), Ctrl+Clickでスクリプトを実行 (executing scripts using Ctrl+Click), ブロックの形をカスタマイズ (customizing block shapes), 自動保存を無効化 (disable autosave), ファイルをドロップ (drop files), 右クリックメニューを色付け (color context menus), エディター内でメッセージ数表示 (showing message count in editor), ステージを左側に表示 (displaying stage on the left), 実行中のブロック表示 (showing executing block), ブロックの色をカスタマイズ (customizing block color), 正確な数を表示 (showing exact number), 非共有プロジェクトを注目のプロジェクトに設定 (setting unshared projects as featured project), 複製したブロックの実行禁止 (banning execution of duplicated blocks), SVGアップロードの修正 (fixing svg upload), フォーラム引用のカスタマイズ (customizing forum quotes), フォーラムのタイムゾーン表示変更 (changing forum timezone display), 表示域拡大 (expanding view), ブロックパレット隠し (hiding block palette), 新しいスプライトの位置を変更 (changing new sprite positions), 注目のプロジェクトを動かす (running featured projects), 拡張機能の自動追加 (auto-adding extensions), 「私が取り組んでいること」文字数制限増加 (increasing what i'm doing char count limit), 投稿画面のアイコン変更 (changing posting screen icons), ステージを録画 (recording stage), リンク化 (linkifying), アイコンにメッセージ数を表示 (displaying message count on icon), プロジェクトをミュート (muting project), ocular.jeffalo.netステータスとリアクションを表示する (showing ocular status and reactions), モバイル版フォーラムを自動リダイレクト (auto-redirect mobile forums), ステージの枠の丸み除去 (removing stage border rounding), スプライト削除確認 (confirming sprite deletion), コメント欄のサイズ変更 (changing comment box size), Scratch 3.0化 (scratch 3.0-ifying), BBCodeを表示 (show bbcode), YouTubeリンク修正 (fixing youtube links)

**Comment by apple502j** [comment]
And I'm not sure if it's possible to convert all the phrases to nouns. i'd like to, tho.

**Comment by Hans5958** [comment]
Note that #2799 won't resolve point 3, so this should be made in a separate PR, or should be discussed now if we need it in there.

By the way, could we have reactions below about what do you think about this. **Thumbs up if you wanted to use title case, and thumbs down if you don't.**

**Comment by Hans5958** [comment]
@mxmou From the PR:

> It would cause reading the list of addons to take longer because you stop for a moment at Each Capitalized Word.

It's really weird that you think so. I don't stop for a moment to read for example a news title, or a page title, or a book title, or anything that uses a title case, let alone when I'm reading the list on the pictures above.

You know what? I just realised something. What if it is good? This seems plausible, but this may give them more attention when turning on addons, especially when something like the Alt+GreenFlag monstrosity. (Not sure if this holds true on the addons tab, which implementation is a mistake, just like ST implements easier follower invitation on the studio)

PS. By the way, have we get rid that monstrosity on your PR?

**Comment by Secret-chest** [comment]
@Hans5958 Imagine reading Better Forum Post Editor Buttons, Studio Manager Tools, Editor Dark Mode, etcetera. It feels less natural.

**Comment by Hans5958** [comment]
> Imagine reading Better Forum Post Editor Buttons, Studio Manager Tools, Editor Dark Mode, etcetera. It feels less natural.

This may be your opinion, but for me it is more natural. Part of this because the similar implementation used title case, while SA don't (again, look up at the image examples). 

For example, I used the FancyZones feature on the Microsoft PowerToys, but not the Shortcut Guide. Or maybe that Group Policy Editor, or that Snipping Tool, or even that Command Prompt. We're not even talking about brands such as YouTube with it's YouTube Music or Instagram with it's Instagram Stories.

There are countless of examples that title case is used, and we're not talking about general cases of title case (such as text titles). So, yeah, that's my opinion.

**Comment by DNin01** [comment]
I can't decide whether I like title case or not. I tried it by adding `text-transform: capitalize;` to the CSS and it looks... more professional?

**Comment by Secret-chest** [comment]
> > Imagine reading Better Forum Post Editor Buttons, Studio Manager Tools, Editor Dark Mode, etcetera. It feels less natural.
> 
> This may be your opinion, but for me it is more natural. Part of this because the similar implementation used title case, while SA don't (again, look up at the image examples). 
> 
> For example, I used the FancyZones feature on the Microsoft PowerToys, but not the Shortcut Guide. Or maybe that Group Policy Editor, or that Snipping Tool, or even that Command Prompt. We're not even talking about brands such as YouTube with it's YouTube Music or Instagram with it's Instagram Stories.
> 
> There are countless of examples that title case is used, and we're not talking about general cases of title case (such as text titles). So, yeah, that's my opinion.

But our list is very long. It's much harder to scan.

**Comment by DNin01** [comment]
I might enjoy a new naming system with more noun and subject names because they make the addons feel more like apps, with app-like names. Even Microsoft PowerToys has fun feature names.

But I also feel like they're not that bad the way they are - as descriptive names. We don't have to be like everyone else, after all, and the current naming is... maybe easier to work with.

**Comment by Hans5958** [comment]
> Scratch Addons was made with names that always summarize their addons in mind since that's how most of the addons are named. It may be difficult to change some of those names, such as "Do not automatically run duplicated blocks". These self-explanatory names help users find addons they like. Changing some names in this way may have user-experience consequences as the addon names have less to do with their features. Take "more links", for example.

Then the problem lies on the strict dependence of the name and the wish of it being descriptive. These addon names should be related to the addon itself, yes, but not to the point that it is (a) too long to be memorable/easy to refer it back or (b) discourage people to explore the feature themselves for their own adjustment.

**Comment by BroJac5246** [comment]
> > Scratch Addons was made with names that always summarize their addons in mind since that's how most of the addons are named. It may be difficult to change some of those names, such as "Do not automatically run duplicated blocks". These self-explanatory names help users find addons they like. Changing some names in this way may have user-experience consequences as the addon names have less to do with their features. Take "more links", for example.
> 
> Then the problem lies on the strict dependence of the name and the wish of it being descriptive. These addon names should be related to the addon itself, yes, but not to the point that it is (a) too long to be memorable/easy to refer it back or (b) discourage people to explore the feature themselves for their own adjustment.

I would argue that the name being (reasonably) descriptive is more important because otherwise what's the point? It doesn't do anything useful if the user can't understand it.

**Comment by DNin01** [comment]
> > > Scratch Addons was made with names that always summarize their addons in mind since that's how most of the addons are named. It may be difficult to change some of those names, such as "Do not automatically run duplicated blocks". These self-explanatory names help users find addons they like. Changing some names in this way may have user-experience consequences as the addon names have less to do with their features. Take "more links", for example.
> > 
> > Then the problem lies on the strict dependence of the name and the wish of it being descriptive. These addon names should be related to the addon itself, yes, but not to the point that it is (a) too long to be memorable/easy to refer it back or (b) discourage people to explore the feature themselves for their own adjustment.
> 
> I would argue that the name being (reasonably) descriptive is more important because otherwise what's the point? It doesn't do anything useful if the user can't understand it.

They can still search by descriptions of addons, which, along with category filters, I think is the best way to find addons you want.

**Comment by DNin01** [comment]
> but not to the point that it is (a) too long to be memorable/easy to refer it back

That's a really good point. You probably wouldn't prefer to have to say "enable the 'Do not automatically space overlapping scripts' addon" when it could be shortened to "enable 'No Script Bumping'".

**Comment by BroJac5246** [comment]
> > but not to the point that it is (a) too long to be memorable/easy to refer it back
> 
> That's a really good point. You probably wouldn't prefer to have to say "enable the 'Do not automatically space overlapping scripts' addon" when it could be shortened to "enable 'No Script Bumping'".

Yeah but say you tell me to enable "No Script Bumping." I'll be like "What in the world is that?". But if you tell me to enable "Do not automatically space overlapping scripts" then I'll know immediately what it does. In other words, the first one is so short that it is truly pointless for anything other than referring to the addon– not ideal when the only thing shown when browsing the settings page is the addon name. While it might be easier to mention the addon, the name loses most of its meaning. And if we try to _reasonably_ shorten long names then the issue about referring to the addon will be partially mitigated. We could also _consider_ using contractions to further shorten names, although that's less professional and I'm more split on that.

**Comment by DNin01** [comment]
> > > but not to the point that it is (a) too long to be memorable/easy to refer it back
> > 
> > 
> > That's a really good point. You probably wouldn't prefer to have to say "enable the 'Do not automatically space overlapping scripts' addon" when it could be shortened to "enable 'No Script Bumping'".
> 
> Yeah but say you tell me to enable "No Script Bumping." I'll be like "What in the world is that?". But if you tell me to enable "Do not automatically space overlapping scripts" then I'll know immediately what it does.

If I was talking in the context of Scratch Addons, you'd know to search for the addon by name and find out what it does.

**Comment by Secret-chest** [comment]
Correct title case doesn't capitalise some certain short words (prepositions 3 letters or less, etc.):

**Correct:** Editor Dark Mode and Customizable Colors

**Incorrect:** Editor Dark Mode And Customizable Colors

And some languages such as Romanian don't use title case, even though they are Latin amd have capitals.

I don't like title case, since it makes scanning the compact list much harder: Which of these is better?

* Editor dark mode and customizable colors
* Pause button
* Insert blocks by name
* Studio manager tools
* Cloud games
* 60FPS project player mode

_or_

* Editor Dark Mode and Customizable Colors
* Pause Button
* Insert Blocks by Name
* Studio Manager Tools
* Cloud Games
* 60FPS Project Player Mode

The second list is much slower to read. We're trying to describe the feature, not make a headline.

**Comment by BroJac5246** [comment]
> Correct title case doesn't capitalise some certain short words (prepositions 3 letters or less, etc.):
> 
> **Correct:** Editor Dark Mode and Customizable Colors
> 
> **Incorrect:** Editor Dark Mode And Customizable Colors
> 
> And some languages such as Romanian don't use title case, even though they are Latin and have capitals.
> 
> I don't like title case, since it makes scanning the compact list much harder: Which of these is better?
> 
> * Editor dark mode and customizable colors
> * Pause button
> * Insert blocks by name
> * Studio manager tools
> * Cloud games
> * 60FPS project player mode
> 
> _or_
> 
> * Editor Dark Mode and Customizable Colors
> * Pause Button
> * Insert Blocks by Name
> * Studio Manager Tools
> * Cloud Games
> * 60FPS Project Player Mode
> 
> The second list is much slower to read. We're trying to describe the feature, not make a headline.

I'm saying this as a native English speaker, but I can read both lists at the same speed, and the second one is more correct for a name.

**Comment by DNin01** [comment]
> Correct title case doesn't capitalise some certain short words (prepositions 3 letters or less, etc.):
> 
> **Correct:** Editor Dark Mode and Customizable Colors
> 
> **Incorrect:** Editor Dark Mode And Customizable Colors

Yes, I was thinking about that too.

**Comment by Hans5958** [comment]
A little bit of retrospect: "Editor Dark Mode and Customizxable Colors" can be shortened as "Customizable Colors". But then I would prefer call it as "Themes" and even just do #2232 and we don't have to deal with this long name.

**Comment by DNin01** [comment]
> A little bit of retrospect: "Editor Dark Mode and Customizxable Colors" can be shortened as "Customizable Colors". But then I would prefer calling it as "Themes" and even just do #2232 and we don't have to deal with this long name.

#6009 would help with that a lot.

**Comment by WorldLanguages** [comment]
#6009 suggests that the scope of the addon is made part of the addon name, when ambiguous. When grouping addons by scope, all names will be visibly shorter than they are now.

**Comment by Secret-chest** [comment]
Hey @Hans5958 , if you really love title case that much, why not go make a bot that converts all issue titles to title case?

**Comment by DNin01** [comment]
I think the main reason there are mixed opinions with making the addon names written in title case is because they are more like short descriptions, not proper names. Just look at the screenshot on #6317, it really shows how much more natural names look in title case, but if you try to do the same thing with descriptions, It Does Not Look As Formal.

**Comment by Hans5958** [comment]
> I think the main reason there are mixed opinions with making the addon names written in title case is because they are more like short descriptions, not proper names. Just look at the screenshot on #6317, it really shows how much more natural names look in title case, but if you try to do the same thing with descriptions, It Does Not Look As Formal.

Actually, you have a good point here.

> Hey @Hans5958 , if you really love title case that much, why not go make a bot that converts all issue titles to title case?

Simply put, I'm not talking about issue titles. Also, I can't be bothered on issue titles that is seen by a group of developers compared to addon name which are related to user experience.

**Comment by BroJac5246** [comment]
Let me ask some questions:

Q: What are addon names?
A: They're names, of course. For the addons.

Q: Are names proper nouns?
A: Yes.

Q: So should names be formatted as proper nouns?
A: Duh.

Q: And how do you format proper nouns?
A: You use title case.

So I don't see why we're debating if addon names should use title case. Unless they're not names, in which case... what are they? It's been argued that they're just short descriptions, but, while I think they should be generally _descriptive_, they're not _descriptions_. That's what the, um... descriptions... are for, after all.

**Comment by Secret-chest** [comment]
> Q: Are names proper nouns?
> A: Yes.

I don't think "Website dark mode and customizable colors" is a proper noun. It's not a phrase either, but it's not something that should be considered a proper noun. Title case is for titles. Also take into account translations: in Romanian proper nouns exist, but title case isn't used for titles. "Munții Carpați" is a proper noun, but there's no way "Mod Întunecat pentru Editor și Culori Personalizablie" can be one.

I agree that they are short descriptions, they're not branded app-like names like in PowerToys, BetterDiscord or extension stores, and I like that because it puts discovery first. But I agree names shouldn't be phrases.

**Comment by Samq64** [comment]
I would like to eventually use title case but right now it doesn't really make much sense because a lot of addon names start with adjectives and some are just too long (e.g. Customizable quotes and code blocks on forums) I think it would be best to do it after #6009 but some adjustments to addon names can be done earlier, for example replacing customizable with custom.

**Comment by WorldLanguages** [comment]
> #6009 suggests that the scope of the addon is made part of the addon name, when ambiguous. When grouping addons by scope, all names will be visibly shorter than they are now.

Even with shorter names, I think some addon names will still be too long for title case to look good visually.

**Comment by DNin01** [comment]
> Maybe title case would work if addons got names like "BlockFind", "Ocular Status", "I can't Believe it's Not 2.0", "Better Buttons", "Magic Block Palette".

That's what I was saying.

**Comment by Secret-chest** [comment]
> That's what I was saying.

That's the problem, user experience is more important than marketing.

**Comment by BroJac5246** [comment]
> > That's what I was saying.
> 
> That's the problem, user experience is more important than marketing.

Certainly. But is consistency not part of a good user experience? Like, for example, following a language's formatting rules?

A proper noun is in most cases (including ours) a name. And an addon name is a name. There is no limit on how long a name can be. You could go out and write a book and name it "The Reason why the Quick Brown Fox Decided to Jump over the Lazy Dog." I don't know why you'd do that, but you could. And importantly, it's still a name, and it's still formatted like one.

In the same way, SA's addon _names_ should be formatted as _names_. It doesn't matter how long they are. They're names. Unless they're not names, which has been proposed, but in that case, what are they?

**Comment by Secret-chest** [comment]
> Certainly. But is consistency not part of a good user experience? Like, for example, following a language's formatting rules?

The addon names aren't names like an app or a book name.

**Comment by rgantzos** [comment]
> > Certainly. But is consistency not part of a good user experience? Like, for example, following a language's formatting rules?
> 
> The addon names aren't names like an app or a book name.

Then calling them “names” is misleading. You don’t spell names with lowercase last names. Consistency is good, and they should be consistently in title case. But I agree with Samq64 that the names should be updated first.

**Comment by Secret-chest** [comment]
They aren't names!

**Comment by BroJac5246** [comment]
> They aren't names!

So what are they? And why are they called names?

**Comment by DNin01** [comment]
> > Certainly. But is consistency not part of a good user experience? Like, for example, following a language's formatting rules?
> 
> The addon names aren't names like an app or a book name.

They aren't _currently_.

**Comment by BroJac5246** [comment]
> Also, with you two agreeing to make them phrases just so they are one letter shorter, they can no longer be title case.

The reasoning here is backwards (nothing personal, but it is).

First, note that no one has disputed that proper nouns use title case. Your confusion appears to be over whether or not a title is a title based on if it's a noun/noun phrase or not. A phrase is not _eligible to be a title_ because it is a noun or contains a noun phrase. It doesn't work that way.

A title is a noun not because it _contains_ a noun or a noun phrase, but rather because _it is a thing_ (nouns are generally people, places, things, or ideas). A title is an identifier. And that identifier is a thing/noun. So the title is a thing, or a noun, and it's a referring to a work– it's a title. And because it's referring to a _specific_ work (and thus a specific thing), it's a proper noun. Because that's what all titles do (identify specific works), they're all proper nouns.

And since we seem to agree that proper nouns use title case, and I've shown why titles are proper nouns no matter what they say, hopefully this is clear.

TL;DR:
Titles don't have to be noun phrases. Titles are names given to specific works and therefore are proper nouns, which are capitalized 100% of the time.

**Comment by Secret-chest** [comment]
> The reasoning here is backwards (nothing personal, but it is).
> 
> First, before reading this, note that no one has disputed that proper nouns use title case. Your confusion appears to be over whether or not a title is a title based on if it's a noun/noun phrase or not. A phrase is not eligible to be a title because it is a noun or contains a noun phrase. It doesn't work that way.
> 
> A title is a noun not because it contains a noun or a noun phrase, but rather because it is a thing (nouns are generally people, places, things, or ideas). A title is an identifier. And that identifier is a thing/noun. So the title is a thing, or a noun, and it's a referring to a work– it's a title. And because it's referring to a specific work (and thus a specific thing), it's a proper noun. Because that's what all titles do (identify specific works), they're all proper nouns.
> 
> And since we seem to agree that proper nouns use title case, and I've shown why titles are proper nouns no matter what they say, hopefully this is clear.
> 
> TL;DR:
> Titles don't have to be noun phrases. Titles are names given to specific works and therefore are proper nouns, which are capitalized 100% of the time.

Sorry, I'm not a native. What's a noun phrase?

**Comment by TylerMS887** [comment]
> What's a noun phrase?

https://en.wikipedia.org/wiki/Noun_phrase

**Comment by DNin01** [comment]
The more I think about it, I'm like, "do descriptive names work better for _our_ project?" Apps like Microsoft PowerToys don't have nearly as many features as we do, and I think people tend to _explore_ PowerToys more than _searching for something in particular_, like we do in Scratch Addons(?). In that case, making sure the titles of features are extra-descriptive can improve user experience, but having app-like names might boost engagement and encourage exploration? I'm not an expert on user engagement psychology. But that's why I think PowerToys and maybe some other programs do it that way.

**Comment by Hans5958** [comment]
> Addons aren't products, they're toggles on a settings page. [...]
>
> _Originally posted by @mxmou in https://github.com/ScratchAddons/ScratchAddons/pull/6582#pullrequestreview-2132880493_

I disagree. In my opinion, these counts as what we offer, our features, and in a stretch, our "products". I would say that the when they said that they want Scratch Addons, they actually want those features that we offer, in form of the addons. They want dark mode, they want gamepad support, they want opacity slider, and if we can, they want skewing.

If you really want to have this point, this means that we actually should make this names more marketable, even in the expense of descriptiveness. As I said, descriptions should be on the descriptions field, and we already show it by default on the settings page.

**Comment by DNin01** [comment]
Good points.

Choosing a good name can even make the user more interested, which can end up making that item be understood by more people.

I think it would be neat if an addon was called "Find"! As long as there's still enough clues to what it does and it's still easy to find the features you're looking for, what's stopping us from naming them like products?

I thought of a potential UI to solve the problem of names not describing their addons well enough. What if there was an "explore" page where you could see and search through all the addons that we have, and each card would, up-front, have the addon's name, a concise description, and maybe even an icon, similar to PowerToys' Dashboard tab or [this mockup image I made](https://github.com/ScratchAddons/ScratchAddons/assets/106490990/d056aac5-edad-4e9e-a906-a236eab81ed8) for #6317.

**Comment by DNin01** [comment]
I have done just this in #8121. Tell me what you think.

![A mocked-up list of addons showing icons, names, and descriptions](https://github.com/user-attachments/assets/4b1f63d9-6eca-428c-adf5-5db3309948e3)

**Comment by Samq64** [comment]
I don't like a lot of the names. Users shouldn't have to read the description to get a basic idea of what addon does, especially simple ones.

**Comment by Secret-chest** [comment]
Exactly, we don't need catchy names, just for them to be easy to understand.

**Comment by Secret-chest** [comment]
Some of them are OK. Certainly not Parameteroo, Friendly Blocks, One, Two, Run or Code Cam.

Also, users may sometimes want to find an addon causing something, having these names won't help.

**Comment by DNin01** [comment]
I get it now.

**Comment by AeroKoder** [comment]
I think there is four options we could do (Ordered in the way I would personally prefer),

1. Rename all of the addons to all be in title case.
2. Have short descriptions instead of names
3. Just leave them alone and don't worry about it.
4. Both, have a name that is in title case and have a short description that shows up under the name, if you want to view the full description of the addon you can hit a "more" button.

Personally, I think renaming all of the addons to be in title case would be the best option. It is most standard to have names in title case.

**Comment by DNin01** [comment]
Maybe we should break from the idea of addons being products and think of them as either features or settings that control features. For that, I think addon names should be a cross between proper nouns and descriptive titles, as seen in real system settings UIs, which our addon name convention already follows for the most part.

**Comment by DNin01** [comment]
For those who like addons being products, some system settings like "VoiceOver" are more like technologies which are somewhat products, so there's no shame in applying a name like "infinite scrolling" if that makes an addon easier to represent than if it was given a descriptive title.
